Argo/Types/Decodable.swift

```diff
@@ -1,4 +1,4 @@
 public protocol Decodable {
   typealias DecodedType = Self
-  static func decode(json: JSON) -> Decoded<DecodedType>
+  static func decode(json: JSON) throws -> DecodedType
```

Comment:
Yep, that's exactly right. This typealias exists to help with :ghost: INHERITANCE :ghost: